code/game/objects/structures/razorwire.dm

```diff
@@ -58,10 +64,14 @@
 	M.apply_damage(RAZORWIRE_BASE_DAMAGE, BRUTE, def_zone, armor_block, TRUE, updating_health = TRUE)
 	razorwire_tangle(M)
 
-/obj/structure/razorwire/CheckExit(atom/movable/mover, direction)
-	. = ..()
+/obj/structure/razorwire/proc/on_try_exit(datum/source, atom/movable/mover, direction, list/knownblockers)
+	SIGNAL_HANDLER
 	if(CHECK_BITFIELD(mover.flags_pass, PASSSMALLSTRUCT))
-		return TRUE
+		return NONE
+	if(!density || !(flags_atom & ON_BORDER) || !(direction & dir) || (mover.status_flags & INCORPOREAL))
```

Comment:
```suggestion
	if(!mover.density || !(flags_atom & ON_BORDER) || !(direction & dir) || (mover.status_flags & INCORPOREAL))
```

Reply:
and uh, what's with the dir check?

Reply:
legacy code for onborder, this is maintiaining parity